baseplate/context/memcache/__init__.py

```diff
@@ -107,31 +107,147 @@ def __init__(self, context_name, server_span, pooled_client):
-    add = monitored('add')
-    incr = monitored('incr')
-    decr = monitored('decr')
-    touch = monitored('touch')
-    stats = monitored('stats')
-    flush_all = monitored('flush_all')
-    quit = monitored('quit')
+    def close(self):
+        with self._get_span("close"):
+            return self.pooled_client.close()
+
+    def set(self, key, value, expire=0, noreply=None):
+        with self._get_span("set") as span:
+            span.set_tag("key", key)
+            span.set_tag("expire", expire)
+            span.set_tag("noreply", noreply)
+            span.set_tag("value", value)
+            return self.pooled_client.set(
+                key, value, expire=expire, noreply=noreply)
+
+    def set_many(self, values, expire=0, noreply=None):
+        with self._get_span("set_many") as span:
+            span.set_tag("key_count", len(values))
+            span.set_tag("expire", expire)
+            span.set_tag("noreply", noreply)
+            return self.pooled_client.set_many(
+                values, expire=expire, noreply=noreply)
+
+    def replace(self, key, value, expire=0, noreply=None):
+        with self._get_span("replace") as span:
+            span.set_tag("key", key)
+            span.set_tag("expire", expire)
+            span.set_tag("noreply", noreply)
+            span.set_tag("value", value)
+            return self.pooled_client.replace(
+                key, value, expire=expire, noreply=noreply)
+
+    def append(self, key, value, expire=0, noreply=None):
+        with self._get_span("append") as span:
+            span.set_tag("key", key)
+            span.set_tag("expire", expire)
+            span.set_tag("noreply", noreply)
+            span.set_tag("value", value)
+            return self.pooled_client.append(
+                key, value, expire=expire, noreply=noreply)
+
+    def prepend(self, key, value, expire=0, noreply=None):
+        with self._get_span("prepend") as span:
+            span.set_tag("key", key)
+            span.set_tag("expire", expire)
+            span.set_tag("noreply", noreply)
+            span.set_tag("value", value)
+            return self.pooled_client.prepend(
+                key, value, expire=expire, noreply=noreply)
+
+    def cas(self, key, value, cas, expire=0, noreply=None):
+        with self._get_span("cas") as span:
+            span.set_tag("key", key)
+            span.set_tag("value", value)
+            span.set_tag("cas", cas)
+            span.set_tag("expire", expire)
+            span.set_tag("noreply", noreply)
+            return self.pooled_client.cas(
+                key, value, cas, expire=expire, noreply=noreply)
+
+    def get(self, key, **kwargs):
+        with self._get_span("get") as span:
+            span.set_tag("key", key)
+            return self.pooled_client.get(key, **kwargs)
+
+    def get_many(self, keys):
+        with self._get_span("get_many") as span:
+            span.set_tag("key_count", len(keys))
```

Comment:
r2 is actually sending some key names which I believe has been helpful. I'm not sure if we should keep doing this in light of the same desire to avoid private info in tracing data, but it may be nice to have if we think it's safe enough. 

```python
def make_keys_str(keys):
    keys_str = ','.join((keys))
    if len(keys_str) > 100:
        return keys_str[:100] + "..."
    else:
        return keys_str
```